mage_ai/frontend/components/PipelineDetail/ConfigureBlock/index.tsx

```diff
@@ -91,6 +91,30 @@ function ConfigureBlock({
     type: block?.type,
   });
 
+  const handleOnSave = useCallback(() => {
+    onSave({
+      ...blockAttributes,
+      name: blockAttributes?.name || defaultName,
+    });
+  }, [blockAttributes, defaultName, onSave]);
+
+  useEffect(() => {
+    const handleKeyDown = (event) => {
```

Comment:
We don't use `useKeyboardContext` here b/c it doesn't work well with modals - it still captures keyboard shortcuts for the underlying page (even when stopping event propagation).